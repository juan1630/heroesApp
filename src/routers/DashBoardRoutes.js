import React from 'react'
import { Redirect, Route, Switch } from 'react-router'
import { Navbar } from '../components/ui/NavBar'
import { MarvelSreecn } from '../../src/components/marvel/MarvelSreecn'
import { HeroesScreen } from '../../src/components/heroes/HeroesScreen'

import {DcSreen } from '../../src/components/dc/DcSreen';

export const DashBoardRoutes = () => {
    return (
        <>
            <Navbar />
            <div>
                <Switch>
                    <Route exact path="/marvel" component={ MarvelSreecn }  />
                    <Route exact path="/heroe/:heroeid" component={ HeroesScreen }  />
                    <Route exact path="/dc" component={ DcSreen }  />
                    <Redirect to="marvel" />
                </Switch>
            </div>
        </>
    )
}
