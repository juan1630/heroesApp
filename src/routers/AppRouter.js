import React, { useContext } from 'react';
import {AuthContext } from  '../auth/AuthContext';

import {
      BrowserRouter as Router,
      Switch
} from "react-router-dom";

import { LoginScreen } from '../components/login/LoginScreen';
import { DashBoardRoutes } from './DashBoardRoutes';
import  {PrivatesRoutes} from '../routers/PrivatesRoutes'
import PublicRoutes from './PublicRoutes';


export const AppRouter = () => {


    const  { user } = useContext( AuthContext );

    console.log( user );

  return (
        <Router>
          <div>
            <Switch>
                <PublicRoutes 
                    exact
                    path="/login" 
                    component={ LoginScreen } 
                    isAuthenticated={user.logged}
                  />

                <PrivatesRoutes  path="/" 
                                component={ DashBoardRoutes }
                                 isAuthenticated={ user.logged } 
                                />
            </Switch>
          </div>
        </Router>
      );
}
