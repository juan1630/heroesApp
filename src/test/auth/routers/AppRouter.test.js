import React from 'react';
import {shallow } from 'enzyme';
import { AppRouter } from '../../../routers/AppRouter'
import { AuthContext } from '../../../auth/AuthContext';

describe('Pruebas en el app router', () => {
    
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged : false
        }
    }

    
    test('Debe de mostrar el login sino estoy autenticado', () => {
        
        const wrapper = shallow(  
            <AuthContext.Provider  value={ contextValue }>


                <AppRouter />
            </AuthContext.Provider>
         )

            expect( wrapper ).toMatchSnapshot();

    })
    
    test('Debe de mostrar el componente de marvel si esta autenticado', () => {


        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged : true,
                name: 'Juan'
            }
        }
    
        
        const wrapper = shallow(  
            <AuthContext.Provider  value={ contextValue }>


                <AppRouter />
            </AuthContext.Provider>
         )

         console.log( wrapper.html() ) 

        // expect( wrapper.find('.navbar').exists() ).toBe(true);

    })
    
    
})
