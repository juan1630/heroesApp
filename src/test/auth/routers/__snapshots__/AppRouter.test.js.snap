// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el app router Debe de mostrar el componente de marvel si esta autenticado 1`] = `<AppRouter />`;

exports[`Pruebas en el app router Debe de mostrar el login sino estoy autenticado 1`] = `<AppRouter />`;
